### PR TITLE
feat(942300): improve regex-assembly format

### LINKED ISSUE #2943: Dissasemble/Improve data for rule 942300

### PR DESCRIPTION
**Pull request by fzipi**
Fixes #2943.

We can probably add more comments on the intended matching of each of the classes in the file 🤔

### CODE REVIEW

**Review by theseion — CHANGES_REQUESTED** [review]
Looks great! Can you remove the replaced lines and regenerate? I'm not quite happy yet with the output, not sure whether that's due to the left overs.

File: regex-assembly/942300.ra [hunk 1]
```diff
@@ -3,24 +3,57 @@
...
+  ##!=>
+  #
+  --
+  {
+##!<
+
+{{start-comment}}\s?\d+
+
+##!> assemble
+  \b
```
[the remark this answers is not in the document]
Comment (fzipi) [reply regex-assembly/942300.ra:27]: Added.

File: regex-assembly/942300.ra [hunk 2]
```diff
@@ -3,24 +3,58 @@
...
-\bdiv\s+\s*?\w+\(
-\blike\s+\s*?\w+\(
-\bbetween\s+\s*?\w+\(
-\bnot\s+\s*?\w+\(
-\bregexp\s+\s*?\w+\(
-\brlike\s+\s*?\w+\(
-\|\|\s+\s*?\w+\(
-\&\&\s+\s*?\w+\(
+##!> define required-spaces-plus-sql-function-call \s+\w+\(
+##!> define optional-spaces-plus-sql-function-call \s*\w+\(
```
Comment (theseion) [inline regex-assembly/942300.ra:7]: I think this should use the same space logic as the others. I see no reason to treat this differently:
```suggestion
##!> define lazy-optional-spaces-plus-sql-function-call \s*?\w+\(
```

**Review by theseion — CHANGES_REQUESTED** [review]
I think we should consider replacing the `\s*?` with `[^\S]*` to avoid backtracking. But that's something for a separate issue.

File: regex-assembly/942300.ra (hunk 2 above)
Comment (fzipi) [reply regex-assembly/942300.ra:7]: updated in PR to change all occurrences.

**Comment by fzipi** [comment]
> I think we should consider replacing the `\s*?` with `[^\S]*` to avoid backtracking. But that's something for a separate issue.

I think it could be a good idea, let's create a new issue and see there this applies.

Also, adding some of the basic definitions here to a "library" with definitions that can be included anywhere looks appealing now...

**Pull request merged**